tasks/localizationTasks.ts

```diff
@@ -0,0 +1,174 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as gulp from 'gulp';
+import * as process from 'node:process';
+import * as minimist from 'minimist';
+import { spawnSync } from 'node:child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as util from 'node:util';
+import { EOL } from 'node:os';
+import { Octokit } from '@octokit/rest';
+
+type Options = {
+    userName: string;
+    email: string;
+    owner: string;
+    commitSha: string;
+    targetRemoteRepo: string;
+    baseBranch: string;
+};
+
+const localizationLanguages = ['cs', 'de', 'es', 'fr', 'it', 'ja', 'ko', 'pl', 'pt-br', 'ru', 'tr', 'zh-cn', 'zh-tw'];
+const locFiles = ['bundle.l10n.%s.json', 'package.nls.%s.json'];
+
+function getGeneratedLocalizationChanges(diffFilesAndDirectories: string[]): string[] {
+    if (diffFilesAndDirectories.length == 0) {
+        return [];
+    }
+
+    const allPossibleLocalizationFileNames = getAllPossibleLocalizationFileNames();
+    const changedLocFilesOrDirectory = [];
+
+    for (const diffFileOrDirectory of diffFilesAndDirectories) {
+        const stat = fs.statSync(diffFileOrDirectory);
+        if (
+            stat.isFile() &&
+            allPossibleLocalizationFileNames.some((locFileName) => path.basename(diffFileOrDirectory) === locFileName)
+        ) {
+            console.log(`${diffFileOrDirectory} is changed.`);
+            changedLocFilesOrDirectory.push(diffFileOrDirectory);
+        }
+
+        if (stat.isDirectory() && diffFileOrDirectory !== 'l10n') {
+            console.log('l10n is changed.');
+            changedLocFilesOrDirectory.push('l10n');
+        }
+    }
+
+    return changedLocFilesOrDirectory;
+}
+
+function getAllPossibleLocalizationFileNames(): string[] {
+    const files = [];
+    for (const lang of localizationLanguages) {
+        for (const file of locFiles) {
+            files.push(util.format(file, lang));
+        }
+    }
+    // English
+    files.push('bundle.l10n.json');
+    return files;
+}
+
+async function git_diff(args: string[]): Promise<string[]> {
+    const result = await git(['diff'].concat(args));
+    // Line ending from the stdout of git is '\n' even on Windows.
+    return result
+        .replaceAll('\n', EOL)
+        .split(EOL)
+        .map((fileName, _) => fileName.trim())
+        .filter((fileName) => fileName.length !== 0);
+}
+
+async function git(args: string[], printCommand = true): Promise<string> {
```

Comment:
It seems like we always print the command, so maybe removable?

Reply:
No, when we call `git add remote url`, PAT is encoded like this
`https://${parsedArgs.userName}:${pat}@github.com/${parsedArgs.owner}/${parsedArgs.targetRemoteRepo}.git`
I don't want to reveal the PAT in this case